src/presets/vercel.ts

```diff
@@ -27,12 +27,16 @@ export const vercel = defineNitroPreset({
       const buildConfig = generateBuildConfig(nitro);
       await writeFile(buildConfigPath, JSON.stringify(buildConfig, null, 2));
 
+      const systemNodeVersion = process.versions.node.split('.')[0];
```

Comment:
This is smart! However, I think the Node.js version during the build (both locally and the one Vercel/CI) does use, can differ. I'm wondering if they provide an environment variable helper.

Reply:
Hmm, I don't think there's such an environment variable support as of today 🤔 AFAIK, when building a Nuxt 3 project on Vercel (Zero-Config), their virtual environment would pick up the desired Node version either from the project settings or `package.json` file at its init stage: https://vercel.com/docs/concepts/functions/serverless-functions/runtimes/node-js#node.js-version
so reading `process.versions` at build time should tell the right Node version, and the above documentation also mentions the use of `process.versions`.

I'm not too concerned about the local build, as they should be using the same major Node version. For unique environments where it's the concern, they can set the new `vercel.function.runtime` option, as it takes precedence over the auto-detected Node version.

Reply:
What about using `import { getNodeVersion } from '@vercel/build-utils'` helper function?

Right now that is expected to be used from a Builder so it accepts `config` and `meta`, but we can probably make this work without those parameters for use with Build Output API like this use case.

Would that solve the problem?

Reply:
@styfle Thank you! Sounds like a better plan. Are you going to make some changes there?

Reply:
It seems landed on `@vercel/build-utils@6.0.0`. Checking implementation, the utility is using `engines` field from the nearest `package.json` + checking against vercel supported ranges 👌🏼 (if v6 is not being used, you can temporary use `getPackageJSON` from `pkg-types` which is already installed as Nitro dep with same behavior)

Reply:
Since there is no `config` object, we can set the default here: https://github.com/vercel/vercel/pull/9332

Reply:
Thanks @styfle @pi0!
Updated this PR to use `@vercel/build-utils`.